File: Difficulty: Easy/Transform to Sum Tree/transform-to-sum-tree.cpp
/* A binary tree node
struct Node
{
    int data;
    Node* left, * right;
}; */

class Solution {
  public:

    // Convert a given tree to a tree where every node contains sum of values of
    // nodes in left and right subtrees in the original tree
    int dfs(Node* node) {
        if(node==nullptr) return 0;
        if(node->left==nullptr && node->right == nullptr) {
            
            int x = node->data;
            node->data = 0;
            return x;
        }
        int left = dfs(node->left);
        int right = dfs(node->right);
        int y = node->data;
        node->data = left+right;
        return node->data+y;
    }
    void toSumTree(Node *node) {
        // Your code here
        
        dfs(node);
        
    }
};

File: Difficulty: Easy/Transform to Sum Tree/transform-to-sum-tree.java
/* Structure for Tree Node
class Node {
    int data;
    Node left, right;

    // Constructor
    Node(int val) {
        data = val;
        left = null;
        right = null;
    }
};
*/
class Solution {
    
    private int solve(Node root) {
        if(root==null) return 0;
        int left = solve(root.left);
        int right = solve(root.right);
        int sum = left+right;
        int temp = root.data;
        root.data = sum;
        return sum+temp;
    }
    
    public void toSumTree(Node root) {
        // code here
        int ans = solve(root);
        //System.out.println(ans);
    }
}